src/auto_scheduler/transform_step.cc

```diff
@@ -923,5 +958,275 @@ String ComputeRootStepNode::PrintAsPythonAPI(Array<te::Stage>* stages,
   return ss.str();
 }
 
+/********** Primitives adding new stages **********/
+
+// Common part for steps that add new stages
+// (e.g. CacheReadStep, CacheWriteStep, RfactorStep)
+void AddStageModificationSteps(int step_id, const Array<Step>& transform_steps,
+                               Array<Step>* replay_steps) {
```

Comment:
The description and function name could be improved. This function checks if `step_id` is a step that adds a new stage, and puts the step to `replay_steps` if so. Will this function be extended when adding Rfactor? If not, then it seems to me that this function is unnecessary.

Reply:
Yes, and the Rfactor will involve more complicated check. I have a TODO hint behind.

Reply:
Moved the step traverse parts to this fucntion, updated the function name.